apps/snitch_core/priv/repo/seed/shipping.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Snitch.Seed.Shipping do
+  @moduledoc false
+
+  import Ecto.Query
+  import Snitch.Tools.Helper.{Shipment, Zone}
+
+  alias Snitch.Data.Model.{Country, ShippingMethod, State}
+  alias Snitch.Data.Schema.State, as: StateSchema
+  alias Snitch.Data.Schema.ShippingMethod, as: ShippingMethodSchema
+  alias Snitch.Data.Schema.{ShippingCategory, Zone}
+  alias Snitch.Repo
+
+  @shipping_categories ~w(light heavy fragile)
+
+  @zone_manifest %{
+    "apac" => %{zone_type: "C"},
+    "india" => %{zone_type: "S"},
+    "north-india" => %{zone_type: "S"}
+  }
+
+  def seed! do
+    all_categories = [light, _heavy, fragile] = seed_shipping_categories!()
+    all_zones = [apac, india, north_india] = seed_zones!()
+
+    manifest = %{
+      "smuggle" => {[apac], [light]},
+      "priority" => {all_zones, [light, fragile]},
+      "regular" => {[india, north_india], all_categories},
+      "hyperloop" => {[north_india], [light, fragile]}
+    }
+
+    Enum.map(manifest, fn {name, {zones, categories}} ->
+      ShippingMethod.create(%{name: name, slug: name}, zones, categories)
+    end)
+  end
+
+  def seed_shipping_categories! do
```

Comment:
This function shows it will raise in case of bad query, but I don't think the query here is going to be bad, what's your thoughts, at the max this will give [no_return].

Reply:
Come on let me indulge a bit, and let the name have a `!` in the end. Can you guarantee that the call to `shipping_categories_with_manifest/1` will never raise?
The `!` indicates that the output is not guaranteed to be "graceful" -- we don't return `{:ok, term} | {:error, term}` tuples here.